Setup/CoreData.php

```diff
@@ -0,0 +1,121 @@
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2019 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ *
+ */
+
+namespace Nosto\Tagging\Setup;
+
+use Magento\Framework\Setup\ModuleDataSetupInterface;
+use Magento\Customer\Model\Customer;
+use Magento\Eav\Model\Entity\Attribute\SetFactory as AttributeSetFactory;
+use Magento\Customer\Setup\CustomerSetupFactory;
+use Magento\Framework\Exception\LocalizedException;
+use Nosto\Tagging\Helper\Data as NostoHelperData;
+use Nosto\Tagging\Logger\Logger as NostoLogger;
+
+abstract class CoreData
+{
+    private $customerSetupFactory;
+    private $attributeSetFactory;
+    private $logger;
+
+    /**
+     * CoreData constructor.
+     * @param CustomerSetupFactory $customerSetupFactory
+     * @param AttributeSetFactory $attributeSetFactory
+     * @param NostoLogger $logger
+     */
+    public function __construct(
+        CustomerSetupFactory $customerSetupFactory,
+        AttributeSetFactory $attributeSetFactory,
+        NostoLogger $logger
+    ) {
+        $this->customerSetupFactory = $customerSetupFactory;
+        $this->attributeSetFactory = $attributeSetFactory;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @param ModuleDataSetupInterface $setup
+     * @throws LocalizedException
+     * @throws \Zend_Validate_Exception
+     */
+    public function addCustomerReference(ModuleDataSetupInterface $setup)
+    {
+        $customerSetup = $this->customerSetupFactory->create(['setup' => $setup]);
+
+        $customerEntity = $customerSetup->getEavConfig()->getEntityType(Customer::ENTITY);
+        $attributeSetId = (int)$customerEntity->getDefaultAttributeSetId();
+
+        $attributeSet = $this->attributeSetFactory->create();
+        $attributeGroupId = $attributeSet->getDefaultGroupId($attributeSetId);
+
+        $customerSetup->addAttribute(
+            Customer::ENTITY,
+            NostoHelperData::NOSTO_CUSTOMER_REFERENCE_ATTRIBUTE_NAME,
+            [
+                'type' => 'varchar',
+                'label' => 'Nosto Customer Reference',
+                'input' => 'text',
+                'required' => false,
+                'sort_order' => 120,
+                'position' => 120,
+                'visible' => true,
+                'user_defined' => true,
+                'unique' => true,
+                'system' => false,
+            ]
+        );
+
+        $attribute = $customerSetup->getEavConfig()->getAttribute(
+            Customer::ENTITY,
+            NostoHelperData::NOSTO_CUSTOMER_REFERENCE_ATTRIBUTE_NAME
+        );
+
+        $attribute->addData(
+            [
+                'attribute_set_id' => $attributeSetId,
+                'attribute_group_id' => $attributeGroupId,
+                'used_in_forms' => ['adminhtml_customer', 'customer_account_edit'],
+            ]
+        );
+
+        try {
```

Comment:
Please revert this. See comment https://github.com/Nosto/nosto-magento2/pull/453#discussion_r286457337